Service/Category/BatchQueueProcessor.php

```diff
@@ -138,4 +138,13 @@ protected function processFullReindex($storeId, $categoriesPerPage)
             );
         }
     }
+
+    /**
+     * @param array $affectedProductIds
+     * @return void
+     */
+    public function setAffectedProductIds(array $affectedProductIds): void
```

Comment:
Having a getter/setter is much cleaner and follows good OOP principles. 👍 

I'm wondering if we should have the operation push ids onto an array rather than simply set the values. That seems safer so that one operation doesn't potentially overwrite the other?